app/grandchallenge/components/utils.py

```diff
@@ -0,0 +1,74 @@
+from django.core.exceptions import ObjectDoesNotExist

```

Comment:
I won't have time to review this today but this is the first thing that sticks out.

Reply:
Thinking about where to place `reformat_inputs` - I could just move this to the JobPostSerializer (which is the only place this is currently needed), but what this is doing is reformatting the inputs, so  another solution would be to create a new ComponentInterfaceValue serializer that inherits from the `ComponentInterfaceValuePostSerializer` but overrides the `to-representation` method and directly outputs the inputs in the desired format. The JobPostSerializer would then use that serializer instead. That way we could more easily reuse this elsewhere if ever needed. What do you think?

Reply:
If it's not needed anywhere else then it's not worth abstracting, it would be premature. I always think it is best to:

- Use the code once somewhere
- Then, if you have a similar thing, implement it again, possibly duplicating code
- Then, when you have a third thing that needs it start on abstraction.
 
That way you have a good idea of the differing requirements between the three use cases and can more reliably abstract.

Reply:
Ok, that makes sense. I will move it to the JobPostSerializer then.

Reply:
I think this is called the "use-use-reuse" model, can't find good references.